src/auth_endpoints/signup.auth.ts

```diff
@@ -1,5 +1,69 @@
 import { Request, Response } from "express";
+import prismaClient from "../client.js";
+import crypto, { pbkdf2, pbkdf2Sync } from "crypto";
 
 export async function signupAuth(req: Request, res: Response, next: any) {
-	console.log("muvees backend sign up endpoint called");
+	// return an error code if username or password is not provided
+	if (!req.body.username || !req.body.password) {
+		res.status(400).send("faulty username or password");
+
+		return;
+	}
+
+	// return an error code if username contains invalid characters
+	if (/[^a-zA-Z0-9_@$+]/gi.test(req.body.username)) {
+		res.status(400).send("your username username contains invalid characters");
+
+		return;
+	}
+
+	// check if the username is taken by another user
+	const isUserNameTaken = await prismaClient.user.findFirst({
+		where: {
+			username: req.body.username,
+		},
+	});
```

Comment:
The variable name is misleading. What you are doing here is retrieving the username and not checking if it has been taken
```suggestion
	const userName = await prismaClient.user.findFirst({
		where: {
			username: req.body.username,
		},
	});
```